storage/src/main/java/org/apache/kafka/storage/internals/log/LogOffsetMetadata.java

```diff
@@ -65,6 +73,8 @@ private boolean onSameSegment(LogOffsetMetadata that) {
     // compute the number of bytes between this offset to the given offset
     // if they are on the same segment and this offset precedes the given offset
     public int positionDiff(LogOffsetMetadata that) {
+        if (this.segmentBaseOffset == REMOTE_LOG_UNKNOWN_OFFSET || that.segmentBaseOffset == REMOTE_LOG_UNKNOWN_OFFSET)
+            return 1;
```

Comment:
It's a bit hacky to do this here. I was thinking of doing this in DelayedFetch.

```
            if (endOffset.messageOffset != fetchOffset.messageOffset) {
              if (endOffset.messageOnly() || fetchOffset.messageOnly()) {
                 accumulatedSize += 1
              } else if (endOffset.onOlderSegment(fetchOffset)) {
                // Case F, this can happen when the new fetch operation is on a truncated leader
...
            }
```